apps/chat/bots.py

```diff
@@ -22,13 +17,23 @@ def create_conversation(
     experiment_session: Optional[ExperimentSession] = None,
 ) -> Conversation:
     try:
-        return Conversation(
-            prompt_str=prompt_str,
-            source_material=source_material,
-            memory=ConversationBufferMemory(return_messages=True),
-            llm=llm,
-            experiment_session=experiment_session,
-        )
+        if experiment_session and experiment_session.experiment.assistant:
```

Comment:
Would the `if experiment_session` still be needed after you split the prompt builder out? From what I could see, we're always creating the experiment session before sending the user message to the LLM, so I think we can safely assume that the session will exist here.

Reply:
This function (`create_conversation`) is still used by the prompt builder task so we still need the check here for now.

But there are more refactors coming 😉